lib/wordy_cases.rb

```diff
@@ -0,0 +1,44 @@
+class WordyCase < OpenStruct
+  def test_name
+    'test_%s' % description.downcase.tr_s(" ", '_')
+  end
+
+  def workload
+    [
+      "question = '#{input}'",
+      indent(4, assertion),
+    ].join("\n")
+  end
+
+  def skipped
+    index.zero? ? '# skip' : 'skip'
+  end
+
+  private
+
+  def indent(size, lines)
+    lines.split("\n").map do |line|
+      ' ' * size + line
+    end.join("\n")
+  end
+
+  def assertion
+    return error_assertion unless expected
+
+    "assert_equal(#{expected}, WordProblem.new(question).answer)"
+  end
+
+  def error_assertion
+    [
+      "assert_raises ArgumentError do",
+      indent(2, "WordProblem.new(question).answer"),
+      "end",
+    ].join("\n")
+  end
+end
+
+WordyCases = proc do |data|
+  JSON.parse(data)['cases'].map.with_index do |row, i|
+    WordyCase.new(row.merge('index' => i))
```

Comment:
One thing to bear in mind here is that the failure message should show the actual value returned.